packages/components/src/sandbox/index.native.js

```diff
@@ -0,0 +1,252 @@
+					}
+				} );
+			}
+		}
+
+		Array.prototype.forEach.call( document.querySelectorAll( '[style]' ), removeViewportStyles );
+		Array.prototype.forEach.call( document.styleSheets, function( stylesheet ) {
+			Array.prototype.forEach.call( stylesheet.cssRules || stylesheet.rules, removeViewportStyles );
+		} );
+
+		document.body.style.position = 'absolute';
+		document.body.style.width = '100%';
+		document.body.setAttribute( 'data-resizable-iframe-connected', '' );
+
+		sendResize();
+
+		// Resize events can change the width of elements with 100% width, but we don't
+		// get an DOM mutations for that, so do the resize when the window is resized, too.
+		window.addEventListener( 'resize', sendResize, true );
+} )();`;
+
+const style = `
+	body {
+		margin: 0;
+	}
+	html,
+	body,
+	body > div,
+	body > div iframe {
+		width: 100%;
+	}
+	html.wp-has-aspect-ratio,
+	body.wp-has-aspect-ratio,
+	body.wp-has-aspect-ratio > div,
+	body.wp-has-aspect-ratio > div iframe {
+		height: 100%;
+		overflow: hidden; /* If it has an aspect ratio, it shouldn't scroll. */
+	}
+	body > div > * {
+		margin-top: 0 !important; /* Has to have !important to override inline styles. */
+		margin-bottom: 0 !important;
+	}
+`;
+
+function Sandbox( {
+	html = '',
+	providerUrl = '',
+	scripts = [],
+	styles = [],
+	title = '',
+	type,
+} ) {
+	const ref = useRef();
+	const [ width, setWidth ] = useState( 0 );
+	const [ height, setHeight ] = useState( 0 );
+	const [ contentHtml, setContentHtml ] = useState( getHtmlDoc() );
+
+	const windowSize = Dimensions.get( 'window' );
+	const [ isLandscape, setIsLandscape ] = useState(
+		windowSize.width >= windowSize.height
+	);
+	// On Android, we need to recreate the WebView when the device rotates, otherwise it disappears.
+	// For this purpose, the key value used in the WebView will change when the device orientation gets updated.
+	const key = Platform.select( {
+		android: `${ providerUrl }-${ isLandscape ? 'landscape' : 'portrait' }`,
+		ios: providerUrl,
+	} );
+
+	function getHtmlDoc() {
+		// TODO: Use the device's locale
+		const lang = 'en';
+
+		// Put the html snippet into a html document, and update the state to refresh the WebView,
+		// we can use this in the future to inject custom styles or scripts.
+		// Scripts go into the body rather than the head, to support embedded content such as Instagram
+		// that expect the scripts to be part of the body.
+		const htmlDoc = (
+			<html lang={ lang } className={ type }>
+				<head>
+					<title>{ title }</title>
+					<meta
+						name="viewport"
+						content="width=device-width, initial-scale=1"
+					></meta>
+					<style dangerouslySetInnerHTML={ { __html: style } } />
+					{ styles.map( ( rules, i ) => (
+						<style
+							key={ i }
+							dangerouslySetInnerHTML={ { __html: rules } }
+						/>
+					) ) }
+				</head>
+				<body
+					data-resizable-iframe-connected="data-resizable-iframe-connected"
+					className={ type }
```

Comment:
I think on web a few more class names are added to the `type` here: 
https://github.com/WordPress/gutenberg/blob/c9612a177e3b3b6fc03388ffa42ec3cbbae3968d/packages/block-library/src/embed/embed-preview.js#L81-L85

Should we replicate the same here?

Reply:
I might be overlooking something the logic in the web version but I checked the HTML of the iframes and there're no styles defined for these classes, maybe it's being used for other functionality.

`html` tag:
<img width="1231" alt="Screenshot 2021-08-04 at 17 40 04" src="https://user-images.githubusercontent.com/14905380/128211049-39926c19-fde4-4ab9-81b1-d623164fe8f4.png">

`body` tag:
<img width="1233" alt="Screenshot 2021-08-04 at 17 39 47" src="https://user-images.githubusercontent.com/14905380/128211064-91fed3b9-fc2a-4625-8cc8-996c36a2b3e1.png">

In any case, I'll replicate this logic in the native version just in case.

Reply:
Do you mean [this style file](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/sandbox/style.scss) from the sandbox component? If so, I don't think it's necessary because those styles are only used in the iframe component that is only rendered on the web version ([reference](https://github.com/WordPress/gutenberg/blob/a4bc4d19c4f7d495c96bce931a72089cc9713c41/packages/components/src/sandbox/index.js#L244)).

In [this commit](https://github.com/WordPress/gutenberg/pull/33426/commits/d685ca5ce9ddf44a0e35d496c6879cffea7e7c17) I updated the generated styles and I verified with the changes from [the block settings PR](https://github.com/WordPress/gutenberg/pull/33654) that it properly renders as expected when enabling/disabling the "Resize for smaller devices" option 🎊 .


https://user-images.githubusercontent.com/14905380/128547488-5d364fb0-5b11-4b4e-b622-afaa03c90cab.MP4

Reply:
Sorry, I meant [the one](https://github.com/WordPress/gutenberg/blob/d685ca5ce9ddf44a0e35d496c6879cffea7e7c17/packages/block-library/src/embed/style.scss) from embed, but I guess they were not necessary seeing that it works fine now 🎉